tools/sql_extraction/src/test/kotlin/PythonAnalysisTest.kt

```diff
@@ -0,0 +1,66 @@
+package com.google.cloud.sqlecosystem.sqlextraction
+
+import com.google.cloud.sqlecosystem.sqlextraction.output.QueryUsages
+import org.antlr.v4.runtime.CharStreams
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class PythonAnalysisTest {
+    @Test
+    fun `simple string return`() {
+        val program = """
+def test():
+    return "test"
+"""
+
+        val result = analyze(program)
+
+        assertCorrect(result, "test")
+    }
+
+    @Test
+    fun `simple method argument`() {
+        val program = """
+import something
+something.some_method("test")
+"""
+
+        val result = analyze(program)
+
+        assertCorrect(result, "test")
+    }
+
+    @Test
+    fun `literal concatenation`() {
+        val program = """
+import something
+something.some_method("hello" + "world")
+"""
+
+        val result = analyze(program)
+
+        assertCorrect(result, "(helloworld)")
```

Comment:
How come the result is "(helloworld)" and not "helloworld"?

Reply:
For any complex query fragments, I added parentheses around the internal fragments (https://github.com/GoogleCloudPlatform/bigquery-utils/blob/master/tools/sql_extraction/src/main/kotlin/output/QueryFragment.kt#L74-L78).

Would it be a good idea to trim them off at the end of the recursion?

Reply:
I see. That's fine. You can keep it.